src/Schema/Doctrine/DoctrineSchemaGeneratorHelperTrait.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/**
+ * This file is part of MetaModels/core.
+ *
+ * (c) 2012-2019 The MetaModels team.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * This project is provided in good faith and hope to be usable by anyone.
+ *
+ * @package    MetaModels/core
+ * @author     Christian Schiffler <c.schiffler@cyberspectrum.de>
+ * @copyright  2012-2019 The MetaModels team.
+ * @license    https://github.com/MetaModels/core/blob/master/LICENSE LGPL-3.0-or-later
+ * @filesource
+ */
+
+declare(strict_types = 1);
+
+namespace MetaModels\Schema\Doctrine;
+
+use Doctrine\DBAL\Schema\Column;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\DBAL\Types\Type;
+use MetaModels\Information\MetaModelInformationInterface;
+
+/**
+ * This trait provides common used table manipulation tasks.
+ */
+trait DoctrineSchemaGeneratorHelperTrait
+{
+    /**
+     * Obtain the raw table schema for a MetaModel.
+     *
+     * @param DoctrineSchemaInformation     $schema               The schema to populate.
+     * @param MetaModelInformationInterface $metaModelInformation The MetaModel information.
+     *
+     * @return Table
+     */
+    protected function getSchemaForMetaModel(
+        DoctrineSchemaInformation $schema,
+        MetaModelInformationInterface $metaModelInformation
+    ): Table {
+        $rawSchema = $schema->getSchema();
+        if (!$rawSchema->hasTable($metaModelInformation->getName())) {
+            return $rawSchema->createTable($metaModelInformation->getName());
+        }
+
+        return $rawSchema->getTable($metaModelInformation->getName());
+    }
+
+    /**
+     * Update the information for a single column.
+     *
+     * @param Table  $tableSchema The table schema.
+     * @param string $name        The table name.
+     * @param string $typeName    The type name.
+     * @param array  $options     The options.
+     *
+     * @return Column
+     */
+    protected function setColumnData(Table $tableSchema, string $name, string $typeName, array $options = []): Column
+    {
+        if (!$tableSchema->hasColumn($name)) {
+            return $tableSchema->addColumn($name, $typeName, $options);
+        }
+
+        $column = $tableSchema->getColumn($name);
+        $column->setType(Type::getType($typeName));
+        if ([] !== $options) {
```

Comment:
Shouldn't the options not being always applied? What if I want to force having no options?

Reply:
Yes. My questions relates to the case that the injected table schema already has the column defined with some options. What would happen if the option might have changed? 

Maybe it does not happen in the intended use case of the helper trait but at least the behaviour should be documented.

Reply:
I still do not get you on this one.
Where shall the options change?
As soon as the column is in schema, the schema is the master source, as the attribute is schema aware and therefore reflects the desired state.
If you (as third party developer) want to change the options in the schema, it is up to you to register an own schema generator that will override the column data again (registered with lower priority than the stock generator).

You can see the generator concept somewhat like compiler passes being sequentially run.
This trait merely is a helper that abstracts the whole "loop over all tables and all attributes and configure only the ones of type x" into a single "configure column y of type x"